plugin-flex-ts-template-v2/src/feature-library/custom-transfer-directory/custom-components/DirectoryTab.tsx

```diff
@@ -0,0 +1,88 @@
+import { Alert } from '@twilio-paste/core/alert';
+import { Flex } from '@twilio-paste/core/flex';
+import { Spinner } from '@twilio-paste/core/spinner';
+import { withTaskContext, ITask, Actions, Template, templates } from '@twilio/flex-ui';
+import { useState, useRef, useEffect } from 'react';
+import debounce from 'lodash/debounce';
+
+import DirectoryItem from './DirectoryItem';
+import SearchBox from './SearchBox';
+import { StringTemplates } from '../flex-hooks/strings/CustomTransferDirectory';
+import { DirectoryEntry } from '../types/DirectoryEntry';
+
+export interface TransferClickPayload {
+  mode: 'WARM' | 'COLD';
+}
+
+export interface OwnProps {
+  task: ITask;
+  entries: Array<DirectoryEntry>;
+  isLoading: boolean;
+  onTransferClick: (entry: DirectoryEntry, transferOptions: TransferClickPayload) => void;
+}
+
+const DirectoryTab = (props: OwnProps) => {
+  const [filteredDirectory, setFilteredDirectory] = useState([] as Array<DirectoryEntry>);
+
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  // takes the input in the search box and applies it to filter the entry list
+  const onQueueSearchInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // eslint-disable-next-line no-eq-null, eqeqeq
+    if (event.target != null) {
+      filterDirectoryDebounce();
+    }
+  };
+
+  // function to filter the entries and trigger a re-render
+  const filterDirectory = () => {
+    const tempDir = props.entries.filter((entry) => {
+      const searchString = searchInputRef.current?.value.toLocaleLowerCase() || '';
+      return entry.label.toLocaleLowerCase().includes(searchString);
+    });
+
+    setFilteredDirectory(tempDir);
+  };
+
+  const filterDirectoryDebounce = debounce(filterDirectory, 500, { maxWait: 1000 });
+
+  const onTransferEntryClick = (entry: DirectoryEntry) => async (transferOptions: TransferClickPayload) => {
+    props.onTransferClick(entry, transferOptions);
+    Actions.invokeAction('HideDirectory');
+  };
+
+  useEffect(() => {
+    filterDirectory();
+  }, [props.entries]);
+
+  return (
+    <Flex key="external-directory-tab-list" vertical wrap={false} grow={1} shrink={1}>
+      <SearchBox key="key-tab-search-box" onInputChange={onQueueSearchInputChange} inputRef={searchInputRef} />
+      <Flex key="external-tab-results" vertical element="TRANSFER_DIR_COMMON_ROWS_CONTAINER">
+        {props.isLoading && (
+          <Flex hAlignContent="center">
+            <Spinner decorative size="sizeIcon90" />
+          </Flex>
+        )}
+        {filteredDirectory.length === 0 && !props.isLoading ? (
+          <Alert variant="neutral">
+            <Template source={templates[StringTemplates.NoItemsFound]} />
```

Comment:
Would it be possible to parameterize this a little further so we can have tab specific messages when no items are found?  I think it is useful to qualify the likely reason no items are found, i can foresee people raising issues when queues are not available because no one else is logged in etc.

Reply:
Added optional prop to DirectoryTab allowing a custom no entries message. The generic message is used as a fallback if none is provided, as well as for no search results.